packages/codegen-ui-react/lib/forms/form-state.ts

```diff
@@ -81,15 +83,23 @@ export const getDefaultValueExpression = (
  * @returns
  */
 export const getUseStateHooks = (fieldConfigs: Record<string, FieldConfigMetadata>): Statement[] => {
-  const stateNames = new Set<string>();
-  return Object.entries(fieldConfigs).reduce<Statement[]>((acc, [name, { dataType, componentType }]) => {
-    const stateName = name.split('.')[0];
-    if (!stateNames.has(stateName)) {
-      acc.push(buildUseStateExpression(stateName, getDefaultValueExpression(name, componentType, dataType)));
-      stateNames.add(stateName);
-    }
-    return acc;
-  }, []);
+  const stateNameMap = new Map();
+  return Object.entries(fieldConfigs).reduce<Statement[]>(
+    (acc, [name, { dataType, componentType, mappedVariableName }]) => {
+      const stateName = name.split('.')[0];
+      if (!stateNameMap.has(mappedVariableName || stateName)) {
+        acc.push(
+          buildUseStateExpression(
+            mappedVariableName || stateName,
+            getDefaultValueExpression(name, componentType, dataType),
+          ),
+        );
+        stateNameMap.set(mappedVariableName || stateName, mappedVariableName || stateName);
```

Comment:
why the need for a map if key/value are the same?
If map is required can we use a variable to resolve this conditional?

Reply:
I think this was an accident. Meant to keep the Set instead

Reply:
Refactoring to use variable for conditional